src/components/Scenarios/Scenarios.jsx

```diff
@@ -185,7 +185,7 @@ class Scenarios extends React.Component {
           />
           <Heading title={strings.explorer_results} subtitle={`${data.filter(minSampleSize).length} ${strings.explorer_num_rows}`} />
           <Table
-            key={selectedTab}
+            key={selectedTab + JSON.stringify(data && data[0])}
```

Comment:
maybe just don't supply a key?

Reply:
I tried that. It doesn't work.

Reply:
or just generate a GUID?  I don't think data[0] makes a very good key

Reply:
I'm concerned that data[0] might be the same for two different queries, which would cause the table not to update on the new query

Reply:
Wouldn't the table re-render anyway because it receives a new data prop? Not entirely sure how it works.

Reply:
not sure, react may use the key as a shortcut to determine that it should not update, even if the props change.

Reply:
Maybe this is a bug with the Table component and how the pagination works.